src/Documents.jl

```diff
@@ -134,6 +134,31 @@ immutable EvalNode
     result :: Any
 end
 
+# Navigation
+# ----------------------
+
+"""
+Element in the navigation tree of a document, containing navigation references
+to other page, reference to the [`Page`](@ref) object etc.
+"""
+type NavNode
+    page           :: Nullable{Compat.UTF8String} # unless null, should point to a valid Page in the document
```

Comment:
It will anyway, once the code tries to fetch the `Page` item from the `doc`. I think there's any point in checking for this in the constructor. The comment really is just a tip for whoever is trying to figure out what the fields contain exactly, with the "should" meaning "it will most likely do if the code works correctly".